src/Core/Domain/Product/Feature/DTO/FeatureCollection.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Feature\DTO;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Feature\ValueObject\CustomizableFeatureValue;
+use PrestaShop\PrestaShop\Core\Domain\Product\Feature\ValueObject\FeatureValue;
+
+/**
+ * Gets a collection from all possible feature values that can be applied for product.
+ */
+class FeatureCollection
+{
+    /**
+     * @var FeatureValue[]
+     */
+    private $existingFeatureValues;
+
+    /**
+     * @var CustomizableFeatureValue[]
+     */
+    private $customizableFeatureValues;
```

Comment:
Hmm, do we need 2 arrays for product features? 

How about some kind of `ProductFeature` object which could be created either with `ProductFeature::withPredefinedValue($featureId, $featureValueId)` or `ProductFeature::withCustomizedValue($featureId, 'custom value')`?

Reply:
and in once case` ProductFeature->getValue()` will return `id` and in another case `string` ?

Reply:
No, it could have interface which is something like this:
```php
class ProductFeature {
    public function isPredefinedValue(): bool { ... }

    public function getPredefinedValue(): ?FeatureValueId { ... }

    public function getCustomizedValue(): ?string { ... }
}
```

Reply:
im really not a fan of having interface with two possible outcomes - in my opinion its better to have two objects and avoid if conditions thus reducing code complexity.